src/apis/token-service.ts

```diff
@@ -38,35 +44,36 @@ export async function fetchTokenList(chainId: string): Promise<Response> {
  */
 export async function syncTokens(chainId: string): Promise<void> {
   const syncURL = syncTokensURL(chainId);
-  const fetchOptions: RequestInit = {
-    referrer: syncURL,
-    referrerPolicy: 'no-referrer-when-downgrade',
-    method: 'GET',
-    mode: 'cors',
-  };
-  fetchOptions.headers = new window.Headers();
-  fetchOptions.headers.set('Content-Type', 'application/json');
-  await timeoutFetch(syncURL, fetchOptions);
+  queryApi(syncURL);
 }
 
 /**
- * Fetch metadata for the token address provided for a given network chainId
+ * Perform fetch request against the api
  *
- * @return Promise resolving token metadata for the tokenAddress provided
+ * @return Promise resolving request response
  */
-export async function fetchTokenMetadata(
-  chainId: string,
-  tokenAddress: string,
-): Promise<Response> {
-  const tokenMetadataURL = getTokenMetadataURL(chainId, tokenAddress);
+async function queryApi(apiURL: string): Promise<Response> {
   const fetchOptions: RequestInit = {
-    referrer: tokenMetadataURL,
+    referrer: apiURL,
     referrerPolicy: 'no-referrer-when-downgrade',
     method: 'GET',
     mode: 'cors',
   };
   fetchOptions.headers = new window.Headers();
   fetchOptions.headers.set('Content-Type', 'application/json');
-  const tokenResponse = await timeoutFetch(tokenMetadataURL, fetchOptions);
-  return await tokenResponse.json();
+  return await timeoutFetch(apiURL, fetchOptions);
+}
+
+/**
+ * Parse response
+ *
+ * @return Promise resolving request response json value
+ */
+async function parseJsonResponse(apiResponse: Response): Promise<unknown> {
+  const responseObj = await apiResponse.json();
+  // api may return errors as json without setting an error http status code
+  if (responseObj?.error) {
```

Comment:
nit: Is the `?` necessary given we are checking for the props existence in the first place?

Reply:
responseObj could be null, thats valid json
unlikely tho